amf-api-contract/shared/src/main/scala/amf/apicontract/internal/transformation/stages/ExtendsResolutionStage.scala

```diff
@@ -170,7 +170,7 @@ class ExtendsResolutionStage(profile: ProfileName, val keepEditingInfo: Boolean,
       val resolver = TraitTransformer()
 
       // Iterate operations and resolve extends with inherited traits.
-      val traitList = endpoint.operations.flatMap { operation =>
+      val traitList = endpoint.operations.toList.flatMap { operation =>
```

Comment:
Streams are lazy. In this case when using flatMap for endpoint.operations, which is a Stream, the function is only computed for the first operation. This modifies the endPoint variable by applying the trait to the first operation. As the rest of the values of traitList are never used, the function is never computed for the rest of the operations and the trait is never applied to them. A way to solve is using a not lazy class like Lists.

Reply:
Using traitList.toStream.force also works as this will compute all values of the Stream which translates to function being computed for every operation.

Reply:
Nice catch!